data/plugins/autoreload.lua

```diff
@@ -71,7 +96,7 @@ function dirwatch:check(change_callback, ...)
         local info = system.get_file_info(doc.filename or "")
         if info and info.type == "file" and times[doc] ~= info.modified then
           if not doc:is_dirty() and not config.plugins.autoreload.always_show_nagview then
-            reload_doc(doc)
+            core.add_thread(reload_doc_watch, doc, doc)
```

Comment:
```suggestion
            core.add_thread(reload_doc_watch, doc)
```

Reply:
The "doc" needs to be passed twice, the first time as the "weak reference" of the thread and the second time is to be passed as an argument to reload_doc_watch.